### DISCUSSION TITLE
(FEATURE REQUEST) Support for .PSI image format

### DISCUSSION THREAD

**Discussion by Pudentane** [86Box/86Box#4258] [Ideas]
I was very excited to find that 86box actually supports Teledisk (.TD0) images, which even motivated us to accept copy protected images as per future policy of our website. However, it would be 2x more awesome if you implemented support for .PSI files as well. That should definitely make 86box a second to none solution for running copy protected games!

It would also be nice to have support for other formats such as .ANA (Anadisk), .CP2 (Copy II PC), .PDI (PDIWrite) and .TC ( TC (Transcopy), but they are not urgent at this point and pretty much low priority.

**Comment by OBattler** [comment]
I am going to add .PSI support per your request. Of course, this is going to require some additions to the floppy backend (support for alternate bit clocks and ability to tell the backend where precisely to position a sector), but I'll do it.

**Comment by Pudentane** [comment]
Awesome! Thanks!!!

**Comment by OBattler** [comment]
Looking more into it, I see that the .PSI format has some downsides - no concept of a track (it incorrectly assumes that the C and H fields of the sector ID represent the actual cylinder and head the sector is on, but that need not the case, see eg. 2M (kind of a predecessor to XDF)-formatted floppies as an example), and no GAP lengths, which makes reconstructing a MFM surface (which I need to do for 86Box) more difficult.
Maybe I should contact Hampa of PCE about this, and see what can be done.

On a slightly unrelated topic, I also found some drawbacks of the HxC MFM format after implementing the format, and I have made an updated version of the format (and changed the HxC MFM loader to load and output it) that corrects the drawbacks (mostly the fact RPM and data rate should be per-track, not per-floppy, and rounding the number of bitcells to the nearest byte is a bad idea because the track wraparound can be in the middle of a sector, so you'll then get data errors), and contacted the HxC developer with my code, but I haven't heard from them yet.
86Box already supports (but I have not yet committed, as I have another thing pending before I do) both the original and my modified HxC MFM formats, and if push comes to shove, I could release my modified libhxc DLL, then, until PSI is added, Kryoflux and SCP images can be converted to that new MFM format and such images then used in 86Box (already tested, works perfectly with Visi-On).
